src/security/core/tests/common/security_config_test_utils.c

```diff
@@ -395,3 +396,46 @@ char * get_permissions_config(char * grants[], size_t ngrants, bool add_prefix)
   ddsrt_free (config);
   return prefix_data (config_signed, add_prefix);
 }
+
+char * get_governance_config_ex (bool allow_unauth_pp, bool enable_join_ac, DDS_Security_ProtectionKind discovery_protection_kind, DDS_Security_ProtectionKind liveliness_protection_kind,
+    DDS_Security_ProtectionKind rtps_protection_kind, const char * topic_rules, smime_sign_function signer, void *signer_arg, bool add_prefix)
+{
+  struct kvp vars[] = {
+    { "ALLOW_UNAUTH_PP", allow_unauth_pp ? "true" : "false", 1 },
+    { "ENABLE_JOIN_AC", enable_join_ac ? "true" : "false", 1 },
+    { "DISCOVERY_PROTECTION_KIND", pk_to_str (discovery_protection_kind), 1 },
+    { "LIVELINESS_PROTECTION_KIND", pk_to_str (liveliness_protection_kind), 1 },
+    { "RTPS_PROTECTION_KIND", pk_to_str (rtps_protection_kind), 1 },
+    { "TOPIC_RULES", topic_rules != NULL ? topic_rules : get_governance_topic_rule (NULL, false, false, false, false, PK_N, BPK_N), 1 },
+    { NULL, NULL, 0 }
+  };
+  char * config = ddsrt_expand_vars (governance_xml, &expand_lookup_vars, vars);
+  char * config_signed = signer (config, signer_arg);
```

Comment:
It looks me to that `get_governance_config(a,b,c,d,e,f,g)` is equivalent to `get_governance_config_ex(a,b,c,d,e,f,X,NULL,g)` with
```C
char *X(const char *config, void *arg) {
  (void)arg;
  return get_signed_data(config);
}'
```
And similarly for `get_permissions_config_ex`. With `get_signed_data` used only in this particular context, it would make sense to add that unused argumnet to `get_signed_data` to avoid the wrapper function.

Maybe I am overlooking a detail ...

Reply:
Yes, I agree.